tests/src/integration_tests.rs

```diff
@@ -0,0 +1,105 @@
+        account::AccountHash, runtime_args, PublicKey, RuntimeArgs, SecretKey, U512,
+    };
+    
+    const COUNT_KEY: &str = "count";
+    const COUNTER_INC: &str = "counter_inc";
+    const COUNTER_KEY: &str = "counter";
+    
+    pub struct CounterContract{
+        pub context: TestContext,
+        pub account_addr: AccountHash,
+    }
+
+    impl CounterContract{
+        pub fn deploy() -> Self{
+            let public_key: PublicKey = SecretKey::ed25519([1u8; 32]).into();
+            let account_addr = AccountHash::from(&public_key);
+    
+            let mut context = TestContextBuilder::new()
+                .with_public_key(public_key, U512::from(500_000_000_000_000_000u64))
+                .build();
+    
+            let define_session = {
+                let session_code = Code::from("counter-define.wasm");
+                let session_args = runtime_args! {};
+                SessionBuilder::new(session_code, session_args)
+                    .with_address(account_addr)
+                    .with_authorization_keys(&[account_addr])
+                    .build()
+            };
+            context.run(define_session);
+            Self{
+                context,
+                account_addr
+            }
+        }
+
+        pub fn increment_with_endpoint_call(&mut self){
+            let session_code = Code::NamedKey(COUNTER_KEY.to_string(), COUNTER_INC.to_string());
+            let session_args = runtime_args! {};
+            let session = SessionBuilder::new(session_code, session_args)
+                .with_address(self.account_addr)
+                .with_authorization_keys(&[self.account_addr])
+                .build();
+            self.context.run(session);
+        }
+
+        pub fn increment_with_wasm(&mut self)->i32{
+            let session_args = runtime_args! {};
+            let call_code = Code::from("counter-call.wasm");
+            let session = SessionBuilder::new(call_code, session_args)
+                .with_address(self.account_addr)
+                .with_authorization_keys(&[self.account_addr])
+                .build();
+            self.context.run(session);
+            self.get_counter()
+        }
+
+        pub fn get_counter(&self)->i32{
+            self.context
+                .query(
+                    self.account_addr,
+                    &[COUNTER_KEY.to_string(), COUNT_KEY.to_string()],
+                )
+                .unwrap()
+                .into_t()
+                .unwrap()
+        }
+    }
+
+
+    #[test]
+    fn should_deploy_with_counter_zero() {
+        assert_eq!(CounterContract::deploy().get_counter(), 0);
+    }
+
+    #[test]
+    fn should_increment_with_direct_call() {
+        let mut contract = CounterContract::deploy();
+        for expected_value in 1..=3 {
+            // Increment value using an EntryPoint.
+
+            contract.increment_with_endpoint_call();
+            println!("{}, {}",contract.get_counter(), expected_value);
```

Comment:
That should be `assert_eq!(contract.get_counter(), expected_value)`. Tests shouldn't print anything.